ext/gettext/gettext.c

```diff
@@ -191,7 +192,28 @@ PHP_FUNCTION(bindtextdomain)
 	PHP_GETTEXT_DOMAIN_LENGTH_CHECK(1, ZSTR_LEN(domain))
 
 	if (dir == NULL) {
-		RETURN_STRING(bindtextdomain(ZSTR_VAL(domain), NULL));
+		btd_result = bindtextdomain(ZSTR_VAL(domain), NULL);
+		if (btd_result == NULL) {
+			/* It's not POSIX compliant, but musl returns
+			 * NULL rather than an implementation-defined
+			 * default directory when no directory was
+			 * previously bound. Technically there is no
+			 * default directory on musl, but the user can
+			 * provide one (or more) via the MUSL_LOCPATH
+			 * environment variable. For lack of a better
+			 * idea, we return MUSL_LOCPATH if it is set,
+			 * even though it may contain multiple paths.
+			 * We use secure_getenv() because the musl docs
+			 * say that MUSL_LOCPATH will be ignored in the
+			 * scenarios where secure_getenv() will return
+			 * NULL rather than the value of the variable. */
+			musl_locpath = secure_getenv("MUSL_LOCPATH");
```

Comment:
This will need to be guarded for MacOS and Windows to build, ideally move the decls for `btd_result` and `musl_locpath` into this guard to prevent unused warnings

Reply:
I added a `./configure` check for `secure_getenv` and the corresponding `#if`. It won't matter on musl, but now a fallback to `getenv()` is there in case in case someone write a new libc with this bug that is also missing `secure_getenv()`.